core/src/main/java/com/linecorp/armeria/internal/client/ClientHttpObjectEncoder.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.client;
+
+import com.linecorp.armeria.client.UnprocessedRequestException;
+import com.linecorp.armeria.common.ClosedSessionException;
+import com.linecorp.armeria.common.HttpObject;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.internal.common.HttpObjectEncoder;
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+
+/**
+ * Converts an {@link HttpObject} into a protocol-specific object and writes it into a {@link Channel}.
+ */
+public interface ClientHttpObjectEncoder extends HttpObjectEncoder {
+
+    /**
+     * Writes an {@link RequestHeaders}.
```

Comment:
Question :
`an` is correct? This is so confusing to me :thinking:

Reply:
It's `a`. 😅